model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -880,31 +915,139 @@ async def create_vllm_bundle(
+                streaming_predict_route="/stream",
+                extra_routes=[
+                    OPENAI_CHAT_COMPLETION_PATH,
+                    OPENAI_COMPLETION_PATH,
+                ],
+                env={},
+            ),
+            metadata={},
+        )
+
         return (
             await self.create_model_bundle_use_case.execute(
                 user,
-                CreateModelBundleV2Request(
-                    name=endpoint_unique_name,
-                    schema_location="TBA",
-                    flavor=StreamingEnhancedRunnableImageFlavor(
-                        flavor=ModelBundleFlavorType.STREAMING_ENHANCED_RUNNABLE_IMAGE,
-                        repository=hmi_config.vllm_repository,
-                        tag=framework_image_tag,
-                        command=command,
-                        streaming_command=command,
-                        protocol="http",
-                        readiness_initial_delay_seconds=10,
-                        healthcheck_route="/health",
-                        predict_route="/predict",
-                        streaming_predict_route="/stream",
-                        extra_routes=[
-                            OPENAI_CHAT_COMPLETION_PATH,
-                            OPENAI_COMPLETION_PATH,
-                        ],
-                        env={},
-                    ),
-                    metadata={},
-                ),
+                create_model_bundle_v2_request,
+                do_auth_check=False,
+                # Skip auth check because llm create endpoint is called as the user itself,
+                # but the user isn't directly making the action. It should come from the fine tune
+                # job.
+            )
+        ).model_bundle_id
+
+    async def create_vllm_multinode_bundle(
+        self,
+        user: User,
+        model_name: str,
+        framework_image_tag: str,
+        endpoint_unique_name: str,
+        num_shards: int,
+        nodes_per_worker: int,
+        quantize: Optional[Quantization],
+        checkpoint_path: Optional[str],
+        chat_template_override: Optional[str],
+    ):
+        leader_command = self._create_vllm_bundle_command(
+            model_name,
+            framework_image_tag,
+            num_shards,
+            quantize,
+            checkpoint_path,
+            chat_template_override,
+            multinode=True,
+            is_leader=True,
+            nodes_per_worker=nodes_per_worker,
+        )
+        worker_command = self._create_vllm_bundle_command(
+            model_name,
+            framework_image_tag,
+            num_shards,
+            quantize,
+            checkpoint_path,
+            chat_template_override,
+            multinode=True,
+            is_leader=False,
+            nodes_per_worker=nodes_per_worker,
+        )
+
+        # These env vars e.g. leader name, lws name, namespace should be filled in by Launch automatically
```

Comment:
maybe clarify that the envvars used as values are provided by Launch. wording was a bit confusing initially